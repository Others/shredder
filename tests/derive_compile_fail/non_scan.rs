extern crate shredder;

use shredder::Scan;

struct NotScan {}

#[derive(Scan)]
struct Test0 {
    field: NotScan
}

fn main() {}
