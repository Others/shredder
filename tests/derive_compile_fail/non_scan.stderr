error[E0277]: the trait bound `NotScan: shredder::Scan` is not satisfied
   --> tests/derive_compile_fail/non_scan.rs:7:10
    |
7   | #[derive(Scan)]
    |          ^^^^ the trait `shredder::Scan` is not implemented for `NotScan`
    |
note: required by a bound in `Scanner::<'a>::scan`
   --> src/scan.rs
    |
    |     pub fn scan<T: Scan + ?Sized>(&mut self, from: &T) {
    |                    ^^^^ required by this bound in `Scanner::<'a>::scan`
    = note: this error originates in the derive macro `Scan` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `NotScan: GcDrop` is not satisfied
  --> tests/derive_compile_fail/non_scan.rs:7:10
   |
7  | #[derive(Scan)]
   |          ^^^^ the trait `GcDrop` is not implemented for `NotScan`
   |
note: required by a bound in `check_gc_drop`
  --> src/plumbing.rs
   |
   | pub fn check_gc_drop<T: GcDrop>(_: &T) {}
   |                         ^^^^^^ required by this bound in `check_gc_drop`
   = note: this error originates in the derive macro `Scan` (in Nightly builds, run with -Z macro-backtrace for more info)
